Scripts/models/logit.py

```diff
@@ -299,14 +300,12 @@ def calc_basic_prob(self, impedance):
             Mode (car/transit/bike/walk) : dict
                 Type (time/cost/dist) : numpy 2-d matrix
                     Impedances
-        
-        Returns
-        -------
-        dict
-            Mode (car/transit/bike/walk) : numpy 2-d matrix
-                Choice probabilities
         """
-        return self._calc_prob(self._calc_utils(impedance))
+        self._calc_utils(impedance)
+        self.cumul_dest_prob = {}
+        for mode in self.mode_choice_param:
+            cumsum = self.dest_exps[mode].T.cumsum(axis=0)
+            self.cumul_dest_prob[mode] = cumsum / cumsum[-1]
```

Comment:
`cumsum[-1]` is in principal equal to `self.dest_expsums[mode]["logsum"]` which is used in `_calc_prob()`, but due to floating-point errors, they are not ensured to be exactly equal. Calculating `cumsum / cumsum[-1]` ensures that the last probability in the cumulative probability vector is 1.0 and hence that the outcome of tour destination choice is always an index of the destination vector:

https://github.com/HSLdevcom/helmet-model-system/blob/28ba4bdbf6dfe2c4b968dd6c6e9abe6d3e4e00e8/Scripts/datatypes/tour.py#L148-L150